### ISSUE TITLE
Improve voting process

### ISSUE THREAD

**Issue by Ana06** [openSUSE/osem#2335]
Currently people can see what other people vote. I think that what a person has voted affects (even without realizing) what other people vote. Also, the people voting could share the voting information. One idea to solve this would be that admin and organizers see the result without knowing who voted what. The rest of the people voting only could see their own vote.

**Comment by Ana06** [comment]
@differentreality @ddemaio @bear454 @hennevogel  any opinion?

**Comment by ddemaio** [comment]
@Ana06 I agree it has an affect, and of course there is some significant differences of option for certain votes. I don't know if changing it would have a big difference, but it is worth considering.

**Comment by bear454** [comment]
The blind voting feature hides other users votes during the voting period:
https://github.com/openSUSE/osem/blob/master/app/views/admin/programs/_form.html.haml#L13

**Comment by Ana06** [comment]
@bear454 

> The blind voting feature hides other users votes during the voting period:

But it still show you other's people votes when you have voted and you can change your vote. There is still the privacy concern that I could for example share with other people what another person has voted.

**Comment by hennevogel** [comment]
1. Blind voting should not show what people have voted for, ever. Only show an accumulated vote
1. You should not be able to change your vote after the voting period has ended

**Comment by Ana06** [comment]
@hennevogel 

> 1. Blind voting should not show what people have voted for, ever. Only show an accumulated vote

I think even the accumulated one is a problem and shouldn't be shown as long as I am able to change my own vote. It can also affect what I vote. Let see I think a talk should be in the conference but that is not one of the most important ones. I blindly would vote a 7. But if the accumulated vote is 3, then I may vote 10 to try to compensate the bad votes from other people. After voting, I would see the new accumulated voted, which gives me information about how many people had previously voted and gives me the chance to change my vote again.

> 2\. You should not be able to change your vote after the voting period has ended

At the moment there is not a voting period. That why I think that this feature needs to be rethink. 

What about having a call for voting, handle in a similar way to the other calls (call for paper, booths, etc.)?

- During the call for voting period you are allowed to only see your own vote
- After the call for voting period you are able to see your own vote and the accumulated vote and you are not able to change your vote
- After the call for voting the events appear order by default by the voting result in the admin event page.

@differentreality @ddemaio @bear454 @hennevogel what do you think?

**Comment by bear454** [comment]
for LFNW,  we set a 2-week blind voting period. In our fork I've completely rewritten the Events index, using ajax-datatables-rails, and I only show your votes during the blind voting period. After the 2 weeks, notices start going out, and we collaborate on 'finishing' scoring anything that was left out.

**Comment by Ana06** [comment]
@bear454 then the solution I propose will also work for you, right? That would make easier getting new changes in master.

Using datatables also sounds like a good idea :+1:

**Comment by hennevogel** [comment]
> Blind voting should not show what people have voted for, ever. Only show an accumulated vote

I meant: Only show an accumulated vote after the voting period ended.

**Comment by Ana06** [comment]
@hennevogel 

> I meant: Only show an accumulated vote after the voting period ended.

ok, then it is the same (or similar) to what I suggested. In summary we need a voting period/call for voting or something similar. :wink:

**Comment by bear454** [comment]
We have a voting period; it's defined on the program:

* https://github.com/openSUSE/osem/blob/master/app/views/admin/programs/_form.html.haml#L13-L15
* https://github.com/openSUSE/osem/blob/master/app/views/admin/programs/show.html.haml#L58-L73
